classes/Tools.php

```diff
@@ -641,42 +645,96 @@ public static function getCldr(Context $context = null, $language_code = null)
     }
 
     /**
-    * Return price with currency sign for a given product
-    *
-    * @param float $price Product price
-    * @param object|array $currency Current currency (object, id_currency, NULL => context currency)
-    * @return string Price correctly formated (sign, decimal separator...)
-    * if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
-    */
+     * Return price with currency sign for a given product
+     *
+     * @deprecated Since 1.7.4.0. Please use Locale::formatPrice() instead
+     * @see PrestaShop\PrestaShop\Core\Localization\Locale
+     *
+     * @param float $price
+     *  Product price
+     *
+     * @param object|array $currency
+     *  Current currency (object, id_currency, NULL => context currency)
+     *
+     * @param bool $no_utf8
+     *  Not used anymore
+     *
+     * @param Context|null $context
+     *
+     * @return string Price correctly formatted (sign, decimal separator...)
+     * if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
+     *
+     * @throws LocalizationException
+     */
     public static function displayPrice($price, $currency = null, $no_utf8 = false, Context $context = null)
     {
+        @trigger_error(
+            'Tools::displayPrice() is deprecated since version 1.7.4.0. '
+            . 'Use PrestaShop\PrestaShop\Core\Localization\Locale::formatPrice() instead.',
+            E_USER_DEPRECATED
+        );
+
         if (!is_numeric($price)) {
             return $price;
         }
-        if (!$context) {
-            $context = Context::getContext();
+
+        $context  = $context ?: Context::getContext();
+        $currency = $currency ?: $context->currency;
+
+        if (is_int($currency)) {
+            $currency = Currency::getCurrencyInstance($currency);
         }
-        if ($currency === null) {
-            $currency = $context->currency;
-        } elseif (is_int($currency)) {
-            $currency = Currency::getCurrencyInstance((int)$currency);
+
+        $container = $context->controller->getContainer();
+        if (null === $container) {
+            $container = SymfonyContainer::getInstance();
         }
 
-        $cldr = self::getCldr($context);
+        /** @var LocaleRepository $localeRepository */
+        $localeRepository = $container->get(self::SERVICE_LOCALE_REPOSITORY);
+        $locale           = $localeRepository->getLocale((string)$context->language->locale);
```

Comment:
How about inject an instance of "LocaleRepository" in Context class directly? I'm pretty sure you are already in a Controller at this moment, and you dont need to inject container here.

Reply:
Well, sometimes it is possible that the Tools' saved controller is a fake one, having no container (`init()` not called)
I had to find something else